File: front_pfe/src/app/model/demande.ts
import { DemandeFile } from "./demande-file";

export class Demande {
    id: number;
    filiale: string;
    date: Date;
    poste: string;
    description: string;
    service: string;
    motif: string;
    dateEmb: Date;
    statut: string;
    demande : DemandeFile;
 
 

    constructor(id: number,filiale : string,date : Date,poste : string,description : string,
        service : string,motif : string,dateEmb : Date,statut : string,demande : DemandeFile,) {

            this.id = id;
            this.filiale = filiale;
            this.date = date;
            this.poste = poste;
            this.description = description;
            this.service = service;
            this.motif = motif;
            this.dateEmb = dateEmb;
            this.demande = demande;
            this.statut = statut;
            this.demande = demande;
            
    }


}

File: back_pfe/src/main/java/com/app/entity/Demande.java
package com.app.entity;


import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import java.util.Date;

@Data
@AllArgsConstructor
@Entity
@Getter
@Setter
@ToString

@Table(name= "DEMANDE")

public class Demande {
    @Id
    @GeneratedValue
    private int id;
    private String filiale;
    private Date date;
    private String poste;
    private String description;
    private String service;
    private String motif;
    private Date dateEmb;
    private String statut;
    @ManyToOne
    private FileDB demande;

    public Demande() {
    }


}
